src/logger.rs

```diff
@@ -19,17 +19,37 @@ pub struct StarshipLogger {
     log_level: Level,
 }
 
+fn cleanup_old_log_dir(log_dir: &Path) -> io::Result<()> {
+    if !log_dir.join(".migrated").exists()
+        && let Some(home_dir) = utils::home_dir()
+    {
+        let _ = fs::File::create(log_dir.join(".migrated"))?;
+        #[cfg(target_os = "linux")]
+        {
+            if let Some(xdg_cache_home) = env::var_os("XDG_CACHE_HOME").map(PathBuf::from)
+                && xdg_cache_home == home_dir.join(".cache")
+            {
+                return Ok(());
+            }
+        }
+        fs::remove_dir_all(home_dir.join(".cache/starship"))?;
```

Comment:
This could lead to issue with nushell set-ups based on previous instructions that put the init file there.